### PR TITLE
increase cirq requirement upper bound

### PR DESCRIPTION
**Pull request by richrines1**
(because #210 doesn't affect some pre-release versions of `cirq` 0.14.0)

### CODE REVIEW

**Review by vtomole — APPROVED** [review]

**Review by vtomole — DISMISSED** [review]

**Comment by vtomole** [comment]
We can go even higher on bounding the dev version. https://pypi.org/project/cirq/0.14.0.dev20220203001129/ is the newest that doesn't break our tests.

**Comment by vtomole** [comment]
> Cannot instantiate abstract class "Result" with abstract attributes "data", "measurements" and "params"  [abstract]

The new thing is [ResultDict](https://quantumai.google/reference/python/cirq/study/ResultDict).

File: cirq_superstaq/custom_gates_test.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ def test_acecr_eq() -> None:
     assert cirq_superstaq.AceCR("-+", np.pi) != cirq_superstaq.AceCR("+-", np.pi)
 
     assert cirq_superstaq.AceCR("+-", np.pi) == cirq_superstaq.AceCR("+-", 5 * np.pi)
-    assert cirq_superstaq.AceCR("+-", np.pi) != cirq_superstaq.AceCR("+-", 3 * np.pi)
+    assert cirq_superstaq.AceCR("+-", np.pi) == cirq_superstaq.AceCR("+-", 3 * np.pi)
```
Comment (richrines1) [inline cirq_superstaq/custom_gates_test.py:219]: this was previously only False due to a (now corrected) quirk in `cirq.ops.fsim_gate._canonicalize()` where `_canonicalize(pi+0.1) ==  _canonicalize(3*pi+0.1)` but `_canonicalize(pi) !=  _canonicalize(3*pi)`

do we want it to be False generally? we use `cirq.rx` for the sandwiched gate, so technically `AceCR("+-", π)` and `AceCR("+-", 3π)` should differ by a global phase. cirq would typically make these not equal (e.g. `cirq.rx(π) != cirq.rx(3π)`

alternatively we could replace `cirq.rx` in `_decompose_` with a `cirq.XPowGate` so that the global phase doesn't change between `AceCR(..., ϕ)` and `AceCr(..., ϕ+2π)`

Comment (vtomole) [inline cirq_superstaq/custom_gates_test.py:219]: Why the change?

Comment (richrines1) [reply cirq_superstaq/custom_gates_test.py:219]: this was previously only False due to a (now corrected) quirk in `cirq.ops.fsim_gate._canonicalize()` where `_canonicalize(pi+0.1) ==  _canonicalize(3*pi+0.1)` but `_canonicalize(pi) !=  _canonicalize(3*pi)`

do we want it to be False generally? we use `cirq.rx` for the sandwiched gate, so technically `AceCR("+-", π)` and `AceCR("+-", 3π)` should differ by a global phase. cirq would typically make these not equal (e.g. `cirq.rx(π) != cirq.rx(3π)`

alternatively we could replace `cirq.rx` in `_decompose_` with a `cirq.XPowGate` so that the global phase doesn't change between `AceCR(..., ϕ)` and `AceCr(..., ϕ+2π)`

**Comment by richrines1** [comment]
> > Cannot instantiate abstract class "Result" with abstract attributes "data", "measurements" and "params"  [abstract]
> 
> Replacing `Result` with [ResultDict](https://quantumai.google/reference/python/cirq/study/ResultDict) will solve.

will this work for all `cirq>=0.13.0`?

**Comment by vtomole** [comment]
> will this work for all cirq>=0.13.0?

Sadly no, this was released in `0.14.0`.

File: cirq_superstaq/custom_gates_test.py (hunk 1 above)
Comment (vtomole) [reply cirq_superstaq/custom_gates_test.py:219]: Is the question more generally whether our equality check should take into account global phase?

Comment (vtomole) [reply cirq_superstaq/custom_gates_test.py:219]: Discussed offline: Delaying doing this for later to avoid potentially breaking ongoing experiments.

Comment (richrines1) [reply cirq_superstaq/custom_gates_test.py:219]: > Is the question more generally whether our equality check should take into account global phase?

(basically yes. cirq's equality check does for most gates, but we currently don't for AceCR)

**Comment by richrines1** [comment]
> Let's try https://pypi.org/project/cirq/0.14.0.dev20220126174724/

that one seems to work!

**Review by vtomole — APPROVED** [review]
